### ISSUE TITLE
Volume mapping does not work as intended

### ISSUE THREAD

**Issue by gugahoi** [buildkite-plugins/docker-compose-buildkite-plugin#155]
Since releasing `v2.4.0` with volume mapping in the plugin it seems it cannot map relative folders as docker-compose needs the full path to map via the `-v` flags. Unfortunately it seems like we cannot get the full path via `$(pwd)` in `pipeline.yml` so I am unsure how to make use of the flag if not for using data volumes. Below is an example using docker compose cli:

```bash
$ ls -l docs/
total 40
drwxr-xr-x 2 gus gus 4096 May 15 09:55 development
-rw-r--r-- 1 gus gus  551 Dec 27 15:21 images.md
-rw-r--r-- 1 gus gus  456 Dec 27 15:21 storybook.md
drwxr-xr-x 2 gus gus 4096 Jun  5 09:15 testing
drwxr-xr-x 2 gus gus 4096 Dec 27 15:21 tools
drwxr-xr-x 2 gus gus 4096 May 15 09:55 workflow

$ docker volume ls
DRIVER              VOLUME NAME

$ docker-compose run -v ./docs:/tmp/Downloads --entrypoint ls app -l /tmp/Downloads
total 0

$ docker volume ls
DRIVER              VOLUME NAME
local               bd6d29ba0873a698bb03ecd333aaa1bf76afd769440f4385f1e7def98872c80a
local               docs

$ docker-compose run -v $(pwd)/docs:/tmp/Downloads --entrypoint ls app -l /tmp/Downloads
total 40
drwxr-xr-x 2 node node 4096 May 14 23:55 development
-rw-r--r-- 1 node node  551 Dec 27 04:21 images.md
-rw-r--r-- 1 node node  456 Dec 27 04:21 storybook.md
drwxr-xr-x 2 node node 4096 Jun  4 23:15 testing
drwxr-xr-x 2 node node 4096 Dec 27 04:21 tools
drwxr-xr-x 2 node node 4096 May 14 23:55 workflow
```

How can I achieve relative folder volume mapping with the plugin?

**Comment by lox** [comment]
Does $PWD work?

**Comment by toolmantim** [comment]
It’s a shame there’s no way to say “interpolate this when the job runs” in the pipeline.yml syntax.

For now, we could detect `./` and replace it with `$PWD` at runtime?

Any other ideas @clee?

**Comment by clee** [comment]
### examples

If I have this as my step:

```yml
  - label: example step
    plugins:
      docker-compose#v2.4.0:
        run: foo
        env:
          - TARGET=$PWD
```

Then when I run it, `$TARGET` will equal `$PWD` *on the buildkite agent that handles the upload-pipeline step*. Which will **probably** not be the same as `$PWD` on the agent that runs the docker-compose plugin step, unless you only have one agent or you get lucky with agent assignment.

If we use `$$PWD` in your job definition, the way that [the documentation suggests](https://buildkite.com/docs/agent/v3/cli-pipeline#environment-variable-substitution):

```yml
  - label: example step2
    plugins:
      docker-compose#v2.4.0:
        run: foo
        env:
          - TARGET=$$PWD
```

Then the value of `$TARGET` ends up as `\$PWD` in the final docker-compose command line, instead of just `$PWD`, which means that it doesn't get evaluated.

**Comment by gugahoi** [comment]
@clee thanks for the pointer, will have a think about it. I believe I am being a little picky on this one as I can simply use the mappings in my `docker-compose.yml` file. However since they are really more relevant for my build I was thinking I'd keep that file clean and this plugin feature would have helped, however it does not seem to work as I expected it to so maybe there are other use cases.

Do you have any idea if `--no-interpolate` impacts `parallelism`?

**Comment by toolmantim** [comment]
This is fixed in [v2.4.1 (⛰ Vail)](https://github.com/buildkite-plugins/docker-compose-buildkite-plugin/releases/tag/v2.4.1) 🎉
